sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas-evaluation-result-processor/cas-evaluation-result-processor.ts

```diff
@@ -0,0 +1,29 @@
+import { CASSupplier } from "@sims/sims-db";
+import { ProcessSummary } from "@sims/utilities/logger";
+import { CASAuthDetails } from "@sims/integrations/cas";
+import { CASEvaluationResult } from "../cas-supplier.models";
+import { ProcessorResult } from ".";
+
+/**
+ * Process the result of a student evaluation to determine what
+ * should be execute to ensure this student will have a CAS
+ * supplier and a site code associated.
+ */
+export abstract class CASEvaluationResultProcessor {
+  /**
+   * When implemented in a derived class, execute the process
+   * to associate a CAS supplier and a site code to a student.
+   * @param casSupplier student supplier information from SIMS.
+   * @param evaluationResult evaluation result to be processed.
+   * @param auth authentication token needed for possible
+   * CAS API interactions.
+   * @param summary current process log.
+   * @returns processor result.
+   */
+  abstract process(
+    casSupplier: CASSupplier,
+    evaluationResult: CASEvaluationResult,
+    auth: CASAuthDetails,
```

Comment:
If the logon can happen during the time of API call(create supplier or create site), then we can remove from here.

Reply:
The idea in place was to do the logon once and take advantage of it lasts for 1h and be shared.

Reply:
What about getting the auth from a shared service then?

Reply:
> What about getting the auth from a shared service then?

Please see the answer here, the logon was in place and it was just resued. I would be willing to change it, but I do not see how could it be part of the same PR.
The current effort is already putting effort in a good amount of refactoring. It time allows I will do some refactor, not the ideal one but some improvement to avoid passing it as a parameter.
https://github.com/bcgov/SIMS/pull/3804#discussion_r1809328756